CIScripts/Tests/SNATTest.ps1

```diff
@@ -1,3 +1,350 @@
+        }
+
+        if($VM) {
+            throw "SNAT VHDX was not properly cleaned up! Test FAILED"
+        }
+
+        Write-Host "SNAT VHDX was properly cleaned up! Test succeeded"
+    }
+
+    Write-Host "Running Simple SNAT test"
+
+    Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+
+    . $PSScriptRoot\CommonTestCode.ps1
+
+    # SNAT VM options
+    $SNATMgmtSwitchName = "snat-mgmt"
+    $SNATDiskDir = "C:\snat-vm-image"
+    $SNATDiskPath = $SNATDiskDir + "\snat-vm-image.vhdx"
+    $SNATVMDir = "C:\snat-vm"
+
+    # Some random GUID for vrouter_hyperv.py
+    $NameLen = 14
+    $SNAT_GUID = "d8cf77cb-b9b1-4d7d-ad3c-ef54f417cb5f"
+    $SNATVMName = "contrail-wingw-$SNAT_GUID"
+    $SNATLeftName = "int-$SNAT_GUID".Substring(0, $NameLen)
+    $SNATRightName = "gw-$SNAT_GUID".Substring(0, $NameLen)
+    $SNATVethName = "veth-$SNAT_GUID".Substring(0, $NameLen)
+
+    New-MgmtSwitch -Session $Session -MgmtSwitchName $SNATMgmtSwitchName
+
+    $SNATVeth = New-RoutingInterface -Session $Session `
+        -SwitchName $TestConfiguration.VMSwitchName `
+        -Name $SNATVethName `
+        -IPAddress $SNATConfiguration.VethIP
+
+    # TODO: Remove `ForwardingMAC` when Agent is functional
+    New-SNATVM -Session $Session `
+        -VmDirectory $SNATVMDir -DiskPath $SNATDiskPath `
+        -MgmtSwitchName $SNATMgmtSwitchName `
+        -VRouterSwitchName $TestConfiguration.VMSwitchName `
+        -RightGW $SNATConfiguration.GatewayIP `
+        -LeftGW $SNATConfiguration.ContainerGatewayIP `
+        -ForwardingMAC $SNATVeth.MacAddress `
+        -GUID $SNAT_GUID
+
+    Write-Host "Extracting adapters data..."
+    $PhysicalAdapter = Get-RemoteNetAdapterInformation -Session $Session -AdapterName $TestConfiguration.AdapterName
+    $HNSAdapter = Get-RemoteNetAdapterInformation -Session $Session -AdapterName HNSTransparent
+    $SNATLeft = Get-RemoteVMNetAdapterInformation -Session $Session -VMName $SNATVMName -AdapterName $SNATLeftName
+    $SNATRight = Get-RemoteVMNetAdapterInformation -Session $Session -VMName $SNATVMName -AdapterName $SNATRightName
+    Write-Host "Extracting adapters data... DONE"
+
+    $SNATLeftVif = 102
+    $SNATLeftNh = 102
+    $SNATRightVif = 103
+    $SNATRightNh = 103
+
+    Write-Host "Setting up routing rules..."
+    $Res = Invoke-Command -Session $Session -ScriptBlock {
+        route add $Using:SNATConfiguration.GatewayIP mask 255.255.255.255 $Using:SNATConfiguration.VethIP "if" $Using:SNATVeth.IfIndex | Out-Null
+        $LASTEXITCODE
+    }
+    if ($Res -ne 0) {
+        throw "Setting up routing rules... FAILED"
+    }
+    Write-Host "Setting up routing rules... DONE"
+
+    Write-Host "Setting up ARP rule for GW-veth forwarding..."
+    $Res = Invoke-Command -Session $Session -ScriptBlock {
+        netsh interface ipv4 add neighbor $Using:SNATVeth.IfIndex $Using:SNATConfiguration.GatewayIP $Using:SnatRight.MacAddressWindows | Out-Null
```

Comment:
This and https://github.com/codilime/contrail-windows-tools/pull/35/files#diff-e2bf1555d9512887c5b9e73f2f7f2e5cR259 should probably be done by Agent on initialization. Can we add a TODO comment with a ticket number?